### PR TITLE
REF: Add Wallet Hooks

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ntba9b6qak03vbhq49h7xuuej4

[download apk](http://filestorage.bluewallet.io:1488/1595118646.apk)

File: screen/wallets/add.js [hunk 1]
```diff
@@ -15,433 +14,379 @@ import {
...
-  }
-
-  showAdvancedOptions = () => {
-    Keyboard.dismiss();
-    LayoutAnimation.configureNext(LayoutAnimation.Presets.spring);
-    this.setState({ isAdvancedOptionsEnabled: true });
-  };
+const WalletsAdd = () => {
+  const { colors } = useTheme();
+  const styles = StyleSheet.create({
```
Comment (limpbrains) [inline screen/wallets/add.js:40]: Why are you moving StyleSheet.create declarations inside render functions?
If styles declared outside they will be cached. If inside, styles will be new object on each render and will force all pure components to rerender

Comment (marcosrdz) [reply screen/wallets/add.js:40]: Im trying to provide access to useTheme's hook color object to styles. Im not aware of any other way, unless i have 2 separate Stylesheets, one outside and one inside. How would you approach this?

Comment (marcosrdz) [reply screen/wallets/add.js:40]: I ask cause I'm sure theres a better way. This just happens to be the only one I'm aware of.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/5175tyyqwagjjzcpcpjd215npw

[download apk](http://filestorage.bluewallet.io:1488/1595204228.apk)

File: screen/wallets/add.js (hunk 1 above)
Comment (limpbrains) [reply screen/wallets/add.js:40]: How about declare styles as an array 

```js
const styles = StyleSheet.create({
   view: {
     width: 100,
   },
})

const WalletAdd = () => {
  const { colors } = useTheme();

  return <View style=[styles.view, {color: colors.mainColor}]>blablabla</View> 
}
```

Also if you need condition you can do
```js
return <View style=[styles.static, active && {color: 'red'}]>blablabla</View> 
```

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/wg8t99t2teqjkru03tg0mwgqb8

[download apk](http://filestorage.bluewallet.io:1488/1595255426.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/g7pbqyenxr56qcxhn7m6zdat8c

[download apk](http://filestorage.bluewallet.io:1488/1595259922.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/hg5a3m47dgjtk97f0x3e9tccqr

[download apk](http://filestorage.bluewallet.io:1488/1595270558.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/1wjt439cwh09qz1qwhmtencuy8

[download apk](http://filestorage.bluewallet.io:1488/1595285558.apk)

File: screen/wallets/add.js [hunk 2]
```diff
@@ -15,432 +14,383 @@ import {
...
-          <ActivityIndicator />
-        </View>
-      );
-    }
+  useEffect(() => {
+    AsyncStorage.getItem(AppStorage.LNDHUB)
+      .then(setWalletBaseURI)
+      .catch(() => setWalletBaseURI(''));
+    BlueApp.isAdancedModeEnabled().then(setIsAdvancedOptionsEnabled);
+    setIsLoading(false);
```
Comment (limpbrains) [inline screen/wallets/add.js:138]: isLoading will be changed before isAdvancedOptionsEnabled is resolved, looks like it should wait for it

Comment (Overtorment) [reply screen/wallets/add.js:138]: @marcosrdz all good here?

Comment (marcosrdz) [reply screen/wallets/add.js:138]: I'll make the changes @limpbrains  mentioned

**Comment by Overtorment** [comment]
+ what @limpbrains said

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/0qgb73cbv1vxjd4fn7dht3trec

[download apk](http://filestorage.bluewallet.io:1488/1595793059.apk)

**Comment by marcosrdz** [comment]
> plus what @limpbrains said

Addressed.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zgxnu3a96018hqukuufgvpb5gm

[download apk](http://filestorage.bluewallet.io:1488/1596765692.apk)

**Comment by Overtorment** [comment]
reverted because it broke add-wallet in master for me (ios )

**Pull request merged**